### PR TITLE
fix: replace svcat CLI with oc new-app

### PR DESCRIPTION
**Pull request by psturc**
### Comments

You can verify this change by running the backend test suite against OCP 3.11 with RHMI + RHMDS installed (all tests should pass) or against OCP 4.2 with MDS installed via [mobile-services-installer](https://github.com/aerogear/mobile-services-installer).

**Note:** Not all tests are passing on OCP 4.2, I'm currently investigating why.

### CODE REVIEW

File: common/util/rhmds-api.js [hunk 1]
```diff
@@ -171,11 +171,13 @@ const createPushApp = async name => {
...
-  await exec(`svcat provision ${name} \
-    --class datasync-showcase-server \
-    --plan default -n ${namespace} --wait \
-  `);
+const deployShowcaseServer = async (namespace) => {
+  await exec(`oc new-app --template datasync-showcase-server -n ${namespace}`);
+  await waitFor(async () => {
+    const replicasReady = (await exec(`oc get dc -o jsonpath='{.items[*].status.readyReplicas}' -n ${namespace} | wc -w`)).stdout.replace(/\s/g, '')
+    console.log(`Still waiting for deployment of Data Sync Template. Replicas ready: ${replicasReady}/3`)
+    return replicasReady.includes('3')
```
Comment (psturc) [inline common/util/rhmds-api.js]: 3 replicas: postgresql, mqtt, showcase-server

**Comment by b1zzu** [comment]
:eyes:

File: common/util/rhmds-api.js [hunk 2]
```diff
@@ -171,11 +171,13 @@ const createPushApp = async name => {
...
 
-const deployShowcaseServer = async (name, namespace) => {
-  await exec(`svcat provision ${name} \
-    --class datasync-showcase-server \
-    --plan default -n ${namespace} --wait \
-  `);
+const deployShowcaseServer = async (namespace) => {
+  await exec(`oc new-app --template datasync-showcase-server -n ${namespace}`);
+  await waitFor(async () => {
+    const replicasReady = (await exec(`oc get dc -o jsonpath='{.items[*].status.readyReplicas}' -n ${namespace} | wc -w`)).stdout.replace(/\s/g, '')
```
Comment (b1zzu) [inline common/util/rhmds-api.js]: If you run `npm prettier` it will format better this line

File: common/util/rhmds-api.js [hunk 3]
```diff
@@ -171,11 +171,13 @@ const createPushApp = async name => {
...
+    console.log(`Still waiting for deployment of Data Sync Template. Replicas ready: ${replicasReady}/3`)
```
Comment (b1zzu) [inline common/util/rhmds-api.js]: Maybe we can print a less verbose message
```suggestion
    console.log(`Waiting for showcase server: ${replicasReady}/3`)
```

File: common/util/rhmds-api.js (hunk 1 above)
Comment (b1zzu) [reply common/util/rhmds-api.js]:
```suggestion
    // until postgresql, mqtt and showcase-server are ready
    return replicasReady.includes('3')
```

**Pull request merged**